src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -259,6 +259,36 @@ CHIP_ERROR MdnsContexts::Has(GenericContext * context)
     return CHIP_ERROR_KEY_NOT_FOUND;
 }
 
+#ifdef DEBUG
```

Comment:
nit: within chip we prefer `#if` instead of ifdef, so we guarantee the existence of the condition.

Would it make sense to maybe have a config for mdns debug instead of a global debug flag?

Reply:
| Would it make sense to maybe have a config for mdns debug instead of a global debug flag?

If you’re thinking of something similar to `lwip_debug`, I agree it could be useful.

However, I’d generally prefer a solution like `ChipLogDebug(Discovery, ...)` that can be toggled on or off easily, such as through an environment variable or a build-time define.
This approach would be less intrusive than adding #if ... #endif directives throughout the code and aligns better with our current practices.

Followup ?

Reply:
Or did you mean `just for src/platform/Darwin` ?